flex/paths.py

```diff
@@ -156,7 +156,8 @@ def match_path_to_api_path(path_definitions, target_path, base_path='', global_p
         matches = [(p, r.match(target_path)) for p, r in paths.items()]
         matches_by_group_size = collections.defaultdict(list)
         for path, match in matches:
-            matches_by_group_size[len(match.groups())].append(path)
+            if match:
```

Comment:
With the comment above implemented, this conditional should not be necessary since there should be no entries without a match.